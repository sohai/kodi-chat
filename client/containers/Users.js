import React from 'react';
import { compose, pure } from 'recompact';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import usersSelector from '../store/selectors/users';
import UsersInfo from '../components/UsersInfo';

const propTypes = PropTypes && {};

export const UsersRender = ({ you, contact }) => (
  <UsersInfo you={you} contact={contact} />
);

export const mapStateToProps = state => ({
  you: usersSelector.you(state),
  contact: usersSelector.contact(state)
});

UsersRender.propTypes = propTypes;
UsersRender.displayName = 'Users';
export default compose(connect(mapStateToProps), pure)(UsersRender);
