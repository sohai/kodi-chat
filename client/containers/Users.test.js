/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';

import createStore from '../store';

import { UsersRender, mapStateToProps } from './Users';

const store = createStore();

describe('(Container) Users', () => {
  const state = store.getState();
  const initialProps = mapStateToProps(state);
  const props = {
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(<UsersRender {...props} />, { context: { store } });
    expect(wrapper).toMatchSnapshot();
  });
});
