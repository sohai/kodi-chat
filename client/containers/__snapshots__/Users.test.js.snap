// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) Users should render 1`] = `
<pure(UsersInfo)
  contact={
    Object {
      "available": false,
      "name": "",
      "typing": false,
    }
  }
  you={
    Object {
      "available": false,
      "name": "",
    }
  }
/>
`;

exports[`(Container) Users will receive right props 1`] = `
Object {
  "contact": Object {
    "available": false,
    "name": "",
    "typing": false,
  },
  "you": Object {
    "available": false,
    "name": "",
  },
}
`;
